tests/load/filesystem/test_filesystem_common.py

```diff
@@ -256,3 +258,42 @@ def test_filesystem_destination_passed_parameters_override_config_values() -> No
         bound_config = filesystem_destination.configuration(filesystem_config)
         assert bound_config.current_datetime == config_now
         assert bound_config.extra_placeholders == config_extra_placeholders
+
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="Test it only on Windows")
+def test_windows_unc_path() -> None:
+    config = get_config()
+    config.read_only = True
+
+    unc_path = r"\\localhost\\" + os.path.abspath("tests/common/storages/samples").replace(":", "$")
+    abs_path = os.path.abspath(r"tests/common/storages/samples")
+
+    for bucket_url in [
+        unc_path,
+        "file://" + unc_path,
+        abs_path,
+        abs_path.replace(r"\\", "/"),
+    ]:
+
+        filesystem, _ = fsspec_from_config(config)
+
+        try:
+            all_file_items = list(glob_files(filesystem, bucket_url))
+            expected_files = [
+                "freshman_kgs.csv",
+                "freshman_lbs.csv",
+                "mlb_players.csv",
+                "mlb_teams_2012.csv",
+                "mlb_players.jsonl",
+                "A881_20230920.csv",
+                "A803_20230919.csv",
+                "A803_20230920.csv",
+                "mlb_players.parquet",
+                "taxi.csv.gz",
+                "sample.txt",
```

Comment:
When working with UNC, paths are always full and absolute (to include the IP address of the host), so it requires a bit different test case.